core-tests/shared/src/test/scala/zio/CauseSpec.scala

```diff
@@ -31,6 +31,13 @@ object CauseSpec
             check(causes) { c =>
               assert(c.untraced.traces.headOption, isNone)
             }
+          },
+          zio.test.test("`Cause.foldLeft is stack safe") {
```

Comment:
Confused since the test doesn't use `Cause.foldLeft` 😕

Reply:
Ah, sorry. `failures` is implemented in terms of `foldLeft` so I was just using it that to test it. Previously this would stack overflow.

Reply:
The issue is that `foldLeft` is private so we can't test it directly.

Reply:
`private[zio]`?

Reply:
I don't think it makes sense to change the signature just to test it when we can easily test other methods that are implemented in terms of it. I can always rename the test to "Cause.failures is stack safe" if that would be clearer.

Reply:
I updated the test label to make this more clear.